pkg/apis/eksctl.io/v1alpha5/vpc.go

```diff
@@ -97,6 +97,13 @@ type (
 		ExtraCIDRs []*ipnet.IPNet `json:"extraCIDRs,omitempty"`
 		// for pre-defined shared node SG
 		SharedNodeSecurityGroup string `json:"sharedNodeSecurityGroup,omitempty"`
+		// Automatically add security group rules to and from the default
+		// cluster security group and the shared node security group.
+		// This allows unmanagd nodes to communicate with the control plane
+		// and managed nodes.
+		// Defaults to `true`
+		// +optional
+		ManageDefaultSecurityGroupRules *bool `json:"manageDefaultSecurityGroupRules,omitempty"`
```

Comment:
I don't think users expect that `eksctl` adds ingress rules to their provided SGs by default. Could this be false by default? And can we rename it `ManagedSharedNodeSecurityGroup`?

Reply:
I thought about that, but faced a couple issues:

1. Without adding the rules, the provided security group is basically meaningless because you then have to manually find the created security group and add the rules yourself. The fundamental issue is that the cluster security group ID is not known until the EKS cluster is setup.

2. Making it false by default complicates the code a bit. We would effectively have three conditions instead of two. The default of generating a shared security group and adding the rules automatically, the default of providing the shared security group and NOT having the rules added automatically, and the non-default of providing the security group and enabling the setting to add the rules.

Since the default for the generated shared security group is to add the rules, it seemed like that should be the default for an explicitly specified group. Without adding the rules, specifying the security group doesn't do much other than provide access to unmanaged nodes.

I do agree that it's not great because it breaks current behavior, but it could be argued that the current behavior is faulty (especially when using managed nodes).

If you still think the default should be to not add the rules for an explicitly provided security group, I'd be happy to change that.

Reply:
> I don't think users expect that `eksctl` adds ingress rules to their provided SGs by default. Could this be false by default? And can we rename it `ManagedSharedNodeSecurityGroup`?

As for renaming it. Are you suggesting that it should be a string of a boolean? I assume a bool since we already specify the security group ID elsewhere. If that's the case, the bool being true would indicate that the shared node security group would be managed by eksctl?

If so, that seems like it could be confusing to the user since the implicitly created shared node security group would be managed by default.

Reply:
Err wasn't 100% awake when I wrote that name apparently. Should be `ManageSharedNodeSecurityGroupRules` i.e. to match `SharedNodeSecurityGroup`

Reply:
> Err wasn't 100% awake when I wrote that name apparently. Should be `ManageSharedNodeSecurityGroupRules` i.e. to match `VPC.SharedNodeSecurityGroup`
> 
> EDIT: Maybe something like `CreateManagedUnmanagedSecurityGroupRules`?

Ahh, makes more sense now, but still feels odd for that to be false by default when the current eksctl default is to manage the rules for the group it creates.

This is what I struggled with when trying to define the flag. Anything I came up with seemed long or confusing when I tried to differentiate it from the current default behavior.

It also seemed more symmetrical to have the flag work for both the implicitly created shared group and an explicitly provided one.

Reply:
No, I agree I think `true` by default makes sense here.